File: pset6/readability/readability.py
from cs50 import get_string

def main():
    text = get_string("Text: ")
    letter = count_letter(text)
    word = count_word(text)
    sentence = count_sentence(text)

    L = (100 / word) * letter
    S = (100 / word) * sentence

    index = round(0.0588 * L - 0.296 * S - 15.8)

    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")

def count_letter(text):
    counter = 0
    for letter in text:
        if letter >= 'A' and letter <= 'z':
            counter += 1
    return counter

def count_word(text):
    counter = 0
    for space in text:
        if space == " ":
            counter += 1
    counter += 1
    return counter

def count_sentence(text):
    counter = 0
    for sentence in text:
        if sentence == "." or sentence == "!" or sentence == "?":
            counter += 1
    return counter


if __name__ == "__main__":
    main()

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string ptext);
int count_words(string ptext);
int count_sentences(string ptext);
float finding_L(int pletters, float pwords);
float finding_S(int psentences, float pwords);

int main(void)
{
    string text = get_string("Text: ");
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    int L = finding_L(letters, words);
    int S = finding_S(sentences, words);
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int integer_index=round(index);

    if(integer_index>=16)
    {
        printf("Grade 16+\n");
    }
    else if (integer_index<1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", integer_index);
    }


}

int count_letters(string ptext)
{
    int count=0;
    for (int i=0, len=strlen(ptext); i<len; i++)
    {
        if (ptext[i]>='A' && ptext[i]<='z')
        {
            count++;
        }
    }
    return count;
}

int count_words(string ptext)
{
    int count=0;
    for (int i=0, len=strlen(ptext); i<len; i++)
    {
        if (ptext[i]==' ')
        {
            count++;
        }
    }
     return count+1;
}

int count_sentences(string ptext)
{
    int count=0;
    for (int i=0, len=strlen(ptext); i<len; i++)
    {
        if (ptext[i]=='.' || ptext[i]=='!' || ptext[i]=='?')
        {
            count++;
        }
    }
    return count;
}

float finding_L(int pletters, float pwords)
{
    int result;
    result=(100/pwords) * pletters;
    return result;
}

float finding_S(int psentences, float pwords)
{
    int result;
    result=(100/pwords) * psentences;
    return result;
}